internal/xds/rbac/rbac_engine.go

```diff
@@ -38,7 +38,8 @@ import (
 	"google.golang.org/grpc/status"
 )
 
-var getConnection = transport.GetConnection
+// GetConnection allows tests to inject fake connection.
+var GetConnection = transport.GetConnection
```

Comment:
This is intended for unit testing of the rbac engine itself.

Can your tests that need this be written using a real gRPC client & server?  You do have some testing at this level already; can the rest of the test cases be written this way? 
 _Or_, can you stub out the real RBAC engine instead of injecting stuff into the real one?

Reply:
Sounds good. I added these test cases to end-to-end tests.